File: python/maplestory_openapi/api/dto/ranking/overall_ranking.py
from pydantic import BaseModel


class OverallRankingInfo(BaseModel):
    """종합 랭킹 상세 정보

    date(str): 랭킹 업데이트 일자 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
    ranking(int): 종합 랭킹 순위
    character_name(str): 캐릭터 명
    world_name(str): 월드 명
    class_name(str): 직업 명
    sub_class_name(str): 전직 직업 명
    character_level(int): 캐릭터 레벨
    character_exp(int): 캐릭터 경험치
    character_popularity(int): 캐릭터 인기도
    character_guildname(str): 길드 명
    """
    date: str
    ranking: int
    character_name: str
    world_name: str
    class_name: str
    sub_class_name: str
    character_level: int
    character_exp: int
    character_popularity: int
    character_guildname: str | None


class OverallRanking(BaseModel):
    """종합 랭킹 정보

    ranking(list[OverallRankingInfo]): 종합 랭킹 정보
    """
    ranking: list[OverallRankingInfo]

File: csharp/MapleStory.OpenAPI/Src/Dto/Ranking/OverallRanking.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 종합 랭킹 정보
    /// </summary>
    public class OverallRankingDTO
    {
        /// <summary>
        /// 랭킹 업데이트 일자 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
        /// </summary>
        [JsonProperty("date")]
        public DateTimeOffset Date
        {
            get
            {
                return _date.ToOffset(TimeSpan.FromHours(9));
            }
            set
            {
                _date = value;
            }
        }

        private DateTimeOffset _date;

        /// <summary>
        /// 종합 랭킹 순위
        /// </summary>
        [JsonProperty("ranking")]
        public int Ranking { get; set; }

        /// <summary>
        /// 캐릭터 명
        /// </summary>
        [JsonProperty("character_name")]
        public string CharacterName { get; set; }

        /// <summary>
        /// 월드 명
        /// </summary>
        [JsonProperty("world_name")]
        public string WorldName { get; set; }

        /// <summary>
        /// 직업 명
        /// </summary>
        [JsonProperty("class_name")]
        public string ClassName { get; set; }

        /// <summary>
        /// 전직 직업 명
        /// </summary>
        [JsonProperty("sub_class_name")]
        public string SubClassName { get; set; }

        /// <summary>
        /// 캐릭터 레벨
        /// </summary>
        [JsonProperty("character_level")]
        public int CharacterLevel { get; set; }

        /// <summary>
        /// 캐릭터 경험치
        /// </summary>
        [JsonProperty("character_exp")]
        public long CharacterExp { get; set; }

        /// <summary>
        /// 캐릭터 인기도
        /// </summary>
        [JsonProperty("character_popularity")]
        public int CharacterPopularity { get; set; }

        /// <summary>
        /// 길드 명
        /// </summary>
        [JsonProperty("character_guildname")]
        public string? CharacterGuildName { get; set; }
    }

    /// <summary>
    /// 종합 랭킹 응답 정보
    /// </summary>
    public class OverallRankingResponseDTO : RankingResponseDTO<OverallRankingDTO>
    {

    }
}
